src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -116,32 +116,54 @@ public static void WriteParameterNullChecks(this CodeWriter writer, IReadOnlyCol
         {
             foreach (Parameter parameter in parameters)
             {
-                if (parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue))
+                writer.WriteVariableAssignmentWithNullCheck(parameter.Name, parameter);
+            }
+
+            writer.Line();
+        }
+
+        public static void WriteVariableAssignmentWithNullCheck(this CodeWriter writer, string variableName, Parameter parameter)
+        {
+            // Temporary check to minimize amount of changes in existing generated code
+            var assignToSelf = parameter.Name == variableName;
+            if (parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue))
+            {
+                var defaultValue = parameter.DefaultValue.Value;
+                writer
```

Comment:
would if(assignToSelf) {} else {} be more readable here?

Reply:
Nice catch!